From bb8a797e90d404d21d88ebfd483aaa1eb034b529 Mon Sep 17 00:00:00 2001
From: Deva Ramasubramanian <dramasub@codeaurora.org>
Date: Tue, 10 Sep 2013 21:50:43 -0700
Subject: [PATCH 2/5] mm-video: vidc: Rename V4L2_BUF_FLAG_EOS to
 V4L2_QCOM_BUF_FLAG_EOS

V4L2_BUF_FLAG_EOS was renamed to V4L2_QCOM_BUF_FLAG_EOS in the driver
because it's a vendor extension.  Reflect the change in the component
as well.

Change-Id: Ic13f68b1e37cabdffda1805708d68de23a4e870a

Backport c1c7c12df679018002892a5060b09aeece3c5db1 to CM11.
Retain the existing CM11 encoder and decoder implementations.
---
 mm-video-v4l2/vidc/vdec/src/omx_vdec_hevc.cpp             | 4 ++--
 mm-video-v4l2/vidc/vdec/src/omx_vdec_msm8974.cpp          | 6 +++---
 mm-video-v4l2/vidc/venc/src/video_encoder_device_v4l2.cpp | 4 ++--
 3 files changed, 7 insertions(+), 7 deletions(-)

diff --git a/mm-video-v4l2/vidc/vdec/src/omx_vdec_hevc.cpp b/mm-video-v4l2/vidc/vdec/src/omx_vdec_hevc.cpp
index 284c14d3effd8e2dc4948c5f154c659e08a42b16..b581982c72d1dffa20feb0605611442eb41edcc4 100755
--- a/mm-video-v4l2/vidc/vdec/src/omx_vdec_hevc.cpp
+++ b/mm-video-v4l2/vidc/vdec/src/omx_vdec_hevc.cpp
@@ -5155,7 +5155,7 @@ OMX_ERRORTYPE  omx_vdec::empty_this_buffer_proxy(OMX_IN OMX_HANDLETYPE         h
     int rc;
     unsigned long  print_count;
     if (temp_buffer->buffer_len == 0 || (buffer->nFlags & OMX_BUFFERFLAG_EOS)) {
-        buf.flags = V4L2_BUF_FLAG_EOS;
+        buf.flags = V4L2_QCOM_BUF_FLAG_EOS;
         DEBUG_PRINT_HIGH("\n  INPUT EOS reached \n") ;
     }
     OMX_ERRORTYPE eRet = OMX_ErrorNone;
@@ -6149,7 +6149,7 @@ int omx_vdec::async_message_process (void *context, void* message)
                     omxhdr->nTimeStamp = vdec_msg->msgdata.output_frame.time_stamp;
                     omxhdr->nFlags = omx->m_out_mem_ptr[v4l2_buf_ptr->index].nFlags;
 
-                    if (v4l2_buf_ptr->flags & V4L2_BUF_FLAG_EOS) {
+                    if (v4l2_buf_ptr->flags & V4L2_QCOM_BUF_FLAG_EOS) {
                         omxhdr->nFlags |= OMX_BUFFERFLAG_EOS;
                         //rc = -1;
                     }
diff --git a/mm-video-v4l2/vidc/vdec/src/omx_vdec_msm8974.cpp b/mm-video-v4l2/vidc/vdec/src/omx_vdec_msm8974.cpp
index f09953ee2aca4cf4701d831fb162b552d9ee31cb..f719aba1d4477f1fb1bb8a4110a7303fc0ed1466 100644
--- a/mm-video-v4l2/vidc/vdec/src/omx_vdec_msm8974.cpp
+++ b/mm-video-v4l2/vidc/vdec/src/omx_vdec_msm8974.cpp
@@ -5434,7 +5434,7 @@ OMX_ERRORTYPE  omx_vdec::empty_this_buffer_proxy(OMX_IN OMX_HANDLETYPE         h
     int rc;
     unsigned long  print_count;
     if (temp_buffer->buffer_len == 0 || (buffer->nFlags & OMX_BUFFERFLAG_EOS)) {
-        buf.flags = V4L2_BUF_FLAG_EOS;
+        buf.flags = V4L2_QCOM_BUF_FLAG_EOS;
         DEBUG_PRINT_HIGH("\n  INPUT EOS reached \n") ;
     }
     OMX_ERRORTYPE eRet = OMX_ErrorNone;
@@ -6562,7 +6562,7 @@ int omx_vdec::async_message_process (void *context, void* message)
                     omxhdr->nTimeStamp = vdec_msg->msgdata.output_frame.time_stamp;
                     omxhdr->nFlags = 0;
 
-                    if (v4l2_buf_ptr->flags & V4L2_BUF_FLAG_EOS) {
+                    if (v4l2_buf_ptr->flags & V4L2_QCOM_BUF_FLAG_EOS) {
                         omxhdr->nFlags |= OMX_BUFFERFLAG_EOS;
                         //rc = -1;
                     }
@@ -6594,7 +6594,7 @@ int omx_vdec::async_message_process (void *context, void* message)
                     }
                     if (omxhdr && (v4l2_buf_ptr->flags & V4L2_QCOM_BUF_DROP_FRAME) &&
                             !(v4l2_buf_ptr->flags & V4L2_QCOM_BUF_FLAG_DECODEONLY) &&
-                            !(v4l2_buf_ptr->flags & V4L2_BUF_FLAG_EOS)) {
+                            !(v4l2_buf_ptr->flags & V4L2_QCOM_BUF_FLAG_EOS)) {
                         omx->post_event ((unsigned)NULL,(unsigned int)omxhdr,
                                 OMX_COMPONENT_GENERATE_FTB);
                         break;
diff --git a/mm-video-v4l2/vidc/venc/src/video_encoder_device_v4l2.cpp b/mm-video-v4l2/vidc/venc/src/video_encoder_device_v4l2.cpp
index aaa9ea98d169b8689db1b654c6e77f05bd5d1e1a..beff077caedf0257813f4a08693cd8d2553fe5b9 100644
--- a/mm-video-v4l2/vidc/venc/src/video_encoder_device_v4l2.cpp
+++ b/mm-video-v4l2/vidc/venc/src/video_encoder_device_v4l2.cpp
@@ -301,7 +301,7 @@ void* venc_dev::async_venc_message_thread (void *input)
                 if (v4l2_buf.flags & V4L2_QCOM_BUF_FLAG_CODECCONFIG)
                     venc_msg.buf.flags |= OMX_BUFFERFLAG_CODECCONFIG;
 
-                if (v4l2_buf.flags & V4L2_BUF_FLAG_EOS)
+                if (v4l2_buf.flags & V4L2_QCOM_BUF_FLAG_EOS)
                     venc_msg.buf.flags |= OMX_BUFFERFLAG_EOS;
 
                 if (omx->handle->num_planes > 1 && v4l2_buf.m.planes->bytesused)
@@ -1919,7 +1919,7 @@ bool venc_dev::venc_empty_buf(void *buffer, void *pmem_data_buf, unsigned index,
     buf.length = 1;
 
     if (bufhdr->nFlags & OMX_BUFFERFLAG_EOS)
-        buf.flags = V4L2_BUF_FLAG_EOS;
+        buf.flags = V4L2_QCOM_BUF_FLAG_EOS;
 
     buf.timestamp.tv_sec = bufhdr->nTimeStamp / 1000000;
     buf.timestamp.tv_usec = (bufhdr->nTimeStamp % 1000000);
